packages/rsocket-core/src/RSocketSerialization.js

```diff
@@ -60,6 +60,63 @@ export const JsonSerializers = {
   metadata: JsonSerializer,
 };
 
+// JSON Serializer for Buffer Encoders
+export const JSONBufferSerializer: Serializer<*> = {
+  deserialize: JsonSerializer.deserialize,
+  serialize: (data) => {
+    if (Buffer.isBuffer(data) || data == null) {
+      return data;
+    } else return Buffer.from(JSON.stringify(data));
+  },
+};
+
+export const JSONBufferSerializers = {
+  data: JSONBufferSerializer,
+  metadata: JSONBufferSerializer,
+};
+
+
+/**  Serializer that serializes metadata to Buffer and deserializes composite metada to JSON
+ *  If incoming composite metadata contains multiple objects will deserialize and return an array of objects
+ */
+ export const JSONCompositeMetadataSerializer: Serializer<*> = {
+  deserialize: (metadata) => {
+    let str;
+    if (metadata == null) {
+      return null;
+    } else if (typeof metadata === "string") {
+      str = metadata;
+    } else if (Buffer.isBuffer(metadata)) {
+      const buffer: Buffer = metadata;
+      str = buffer.toString("utf8");
+    } else {
+      const buffer: Buffer = Buffer.from(metadata);
+      str = buffer.toString("utf8");
+    }
+
+    if (str.includes("application/json")) {
```

Comment:
should this be at minimum startsWith? or an equals check? But could you give examples here, not sure being flexible here is a big win.